hail/src/main/scala/is/hail/expr/ir/PCode.scala

```diff
@@ -334,3 +339,90 @@ class PCanonicalLocusCode(val pt: PLocus, val a: Code[Long]) extends PLocusCode
 
   def store(mb: EmitMethodBuilder[_], r: Value[Region], dst: Code[Long]): Code[Unit] = Region.storeAddress(dst, a)
 }
+
+object PCallValue {
+  def apply(pt: PCall, call: Settable[_]): PCallValue = new PCanonicalCallSettable(pt, coerce[Int](call))
+}
+
+abstract class PCallValue extends PValue {
+  def ploidy(): Code[Int]
+
+  def isPhased(): Code[Boolean]
+
+  def forEachAllele(mb: EmitMethodBuilder[_], alleleCode: Value[Int] => Code[Unit]): Code[Unit]
+}
+
+object PCanonicalCallSettable {
+  def apply(sb: SettableBuilder, pt: PCall, name: String): PCanonicalCallSettable =
+    new PCanonicalCallSettable(pt, sb.newSettable[Int](s"${ name }_call"))
+}
+
+class PCanonicalCallSettable(val pt: PCall, call: Settable[Int]) extends PCallValue with PSettable {
+  def get: PCallCode = new PCanonicalCallCode(pt, call)
+
+  def store(pc: PCode): Code[Unit] = call.store(pc.asInstanceOf[PCanonicalCallCode].call)
+
+  def ploidy(): Code[Int] = get.ploidy()
+
+  def isPhased(): Code[Boolean] = get.isPhased()
+
+  def forEachAllele(mb: EmitMethodBuilder[_], alleleCode: Value[Int] => Code[Unit]): Code[Unit] = {
+    val cb = EmitCodeBuilder(mb)
+
+    val call2 = cb.memoize[Int](call >>> 3, "fea_call2")
+    val p = cb.memoize[Int](ploidy(), "fea_ploidy")
+    val j = cb.mb.newLocal[Int]("fea_j")
+    val k = cb.mb.newLocal[Int]("fea_k")
```

Comment:
I get the point here, invoking the method builder directly feels wrong.

That said, I don't feel a `newLocal` wrapper is necessary, since we have `fieldBuilder` and `localBuilder` on `CodeBuilder`.